plugin/pkg/global/resourcereferencemanager/admission.go

```diff
@@ -603,6 +574,45 @@ func (r *ReferenceManager) Admit(ctx context.Context, a admission.Attributes, _
 					}(ncp)
 				}
 
+				wg.Add(len(shootList))
+
+				for _, s := range shootList {
+					shootCloudProfile := gardener.BuildCloudProfileReference(s)
+
+					if s.DeletionTimestamp != nil || shootCloudProfile == nil {
+						wg.Done()
+						continue
+					}
+
+					if (shootCloudProfile.Kind != v1beta1constants.CloudProfileReferenceKindCloudProfile ||
+						shootCloudProfile.Name != cloudProfile.Name) &&
+						!isShootWithDependantNscpfl(s, cloudProfile, relevantNamespacedCloudProfiles) {
```

Comment:
Nit: The following might semantically make more sense, WDYT?

```suggestion
					if (shootCloudProfile.Kind == v1beta1constants.CloudProfileReferenceKindCloudProfile &&
						shootCloudProfile.Name != cloudProfile.Name) ||
						!isShootWithDependantNscpfl(s, cloudProfile, relevantNamespacedCloudProfiles) {
```

Reply:
fine by me. might even be a bit more readable without the brackets.